modules/imgcodecs/test/test_gif.cpp

```diff
@@ -0,0 +1,204 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+TEST(Imgcodecs_Gif, read_gif_multi)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root+"gifsuite/imreadmulti_test/output.gif";
+    vector<cv::Mat> img_vec;
+    cv::imreadmulti(filename, img_vec,0,161);
+    const long unsigned int expected_size=161;
+    EXPECT_EQ(img_vec.size(), expected_size);
+    for(long unsigned int i=0;i<img_vec.size();i++){
+        cv::Mat img=img_vec[i];
+        const string xml_filename=root+"gifsuite/imreadmulti_test/frame"+std::to_string(i+1)+".xml";
```

Comment:
Please use png for reference frames. 161 is to much. I propose to check some key frames and sequence size only.